compiler/src/dotty/tools/dotc/parsing/Scanners.scala

```diff
@@ -345,14 +345,15 @@ object Scanners {
       allowLeadingInfixOperators
       && (  token == BACKQUOTED_IDENT
          || token == IDENTIFIER && isOperatorPart(name(name.length - 1)))
-      && ch == ' '
+      && ch <= ' '
```

Comment:
If this is trying to match all whitespace characters, then it would be clearer to use an `isSpace` method for clarity (we can reuse the existing https://github.com/lampepfl/dotty/blob/ea65338e06142f41f0e68b23250c8e22b0ba8837/compiler/src/dotty/tools/dotc/parsing/xml/Utility.scala#L113-L119), this would also avoid matching non-space characters that precede `' '` in ASCII like `\b`.

The documentation of this method also needs to be updated: it currently states "it is followed _on the same line_ by at least one ' ' and a token that can start an expression."